lib/smart_name.rb

```diff
@@ -35,6 +35,22 @@ def new obj
       end
     end
 
+    def namespaces
+      @@namespaces ||= load_namespaces
+    end
+
+    def load_namespaces
```

Comment:
This gets overriden by the use context, and it is just a hash keyed by the mount point keys, so you can really have any object returned, so this could be a Deck card in the Card context.